ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useRef, useCallback } from 'react';
+import {
+  PopoverRole,
+  PopoverPosition,
+  Popover,
+  IconName,
+  Text,
+  ModalFocus,
+  Box,
+} from '../../component-library';
+import { MenuItem } from '../../ui/menu';
+import {
+  IconColor,
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+
+// note: this is not meant to be merged
+// we should convert MenuItem to TS instead of casting as any
+const TsMenuItem = MenuItem as any;
+
+export const ConnectedAccountsMenu = ({
+  isOpen,
+  anchorElement,
+  onClose,
+}: {
+  isOpen: boolean;
+  anchorElement?: HTMLElement;
+  onClose?: () => void;
+}) => {
+  const t = useI18nContext();
+  const popoverDialogRef = useRef<HTMLDivElement | null>(null);
+
+  const handleKeyDown = useCallback(
+    (event) => {
+      if (
+        event.key === 'Tab' &&
+        popoverDialogRef?.current?.contains(event.target) &&
+        onClose
+      ) {
+        onClose();
+      }
+    },
+    [onClose],
+  );
+
+  return (
+    <Popover
+      className="multichain-connected-accounts-menu__popover"
+      referenceElement={anchorElement}
+      role={PopoverRole.Dialog}
+      position={PopoverPosition.Bottom}
+      offset={[0, 0]}
+      padding={0}
+      isOpen={isOpen}
+      flip
+      preventOverflow
+      isPortal
+    >
+      <ModalFocus
+        restoreFocus
+        initialFocusRef={{ current: anchorElement || null }}
+      >
+        <Box onKeyDown={handleKeyDown} ref={popoverDialogRef}>
+          <TsMenuItem iconName={IconName.SecurityTick}>
```

Comment:
Let's add a data-test-id to each of the menu items. For testing, it will be helpful when we write onClick handlers

Reply:
Done